### PR TITLE
Fix broken Chroma CSS styles

### LINKED ISSUE #22348: Syntax Highlighting in `arc-green` Colors all Identifiers in Orange
### Description

From 1.17.1 to 1.17.2, syntax highlighting when viewing files in the `arc-green` theme changed in such a way that many parts of the code are assigned the same orange color. This is inconsistent with both the `gitea` theme and the way `arc-green` behaves when editing files.

For instance, in screenshot [1], regular identifiers use the regular text color, and special identifiers like function names and built-in types get a special color. In screenshot [2], all of these identifiers get the same orange color. In all of the code files I have looked at, the majority of the text was orange rather than the regular gray text color. This does not happen when editing files [3]. The light theme also uses regular text color for regular identifiers and special colors for special identifiers [4].

Syntax highlighting when viewing files in the `arc-green` theme seems to work the same way in `1.18.0` as it does in `1.17.2`

### Screenshots

[1] 1.17.1 - arc-green - view file
image
[2] 1.17.2 - arc-green - view file
image
[3] 1.17.2 - arc-green - edit file
image
[4] 1.17.2 - gitea - view file
image

### Gitea Version

1.17.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

_No response_

### Browser Version

Chromium Version 108.0.5359.124 (Official Build) Arch Linux (64-bit)

### PR DESCRIPTION
**Pull request by wxiaoguang**
The CSS styles in Gitea themes are out-of-sync of Chroma's styles.

This PR introduces a `chroma-style-diff.go` tool to compare the diff. The missing CSS styles have been added manually. They are left as empty to reduce arguments because there was no color for them before.

And this PR fixes #22348, with just 2 lines changed: `.chroma .kt & .n`, these colors are taken from GitHub.

It's good enough for #22348

image

### CODE REVIEW

**Comment by silverwind** [comment]
Need to adapt stylelint:

```yaml
files: ["**/chroma/*.less", "**/codemirror/*.less", "**/standalone/*.less", "**/console/*.less"]
rules:
  scale-unlimited/declaration-strict-value: null
  block-no-empty: false
```

To fix https://github.com/go-gitea/gitea/issues/22348, we need to change the yellow to light-blue color that is use for plaintext, e.g. near-black on light theme.

**Comment by wxiaoguang** [comment]
I fixed it by `ignoreFiles` and inline `/* stylelint-disable block-no-empty */`, that's clearer.

**Comment by silverwind** [comment]
You probably shouldn't put go code into `web_src`, but in either `build` or `tools`.

**Comment by wxiaoguang** [comment]
> You probably shouldn't put go code into `web_src`, but in either `build` or `tools`.

I really like to related files together. Otherwise developers won't find it.

**Comment by silverwind** [comment]
I will see what I can do based on this PR, will probably push a change or two onto here (hope I have push access to your fork).

**Comment by wxiaoguang** [comment]
> Please invite me as a collaborator to your gitea fork on github

Done

**Comment by silverwind** [comment]
Someone else also noticed the lack of `--ext`, follow https://github.com/stylelint/stylelint/issues/6689.

**Comment by wxiaoguang** [comment]
@silverwind does this PR need more changes?

**Comment by silverwind** [comment]
Yes, sorry I haven't gotten around to it yet.

**Comment by silverwind** [comment]
This almost looks good as a minimal fix, but what is this `override.less` exactly about? Can we remove it? I still prefer to think as these styles as "base" and the themes extend this base.

**Review by silverwind — APPROVED** [review]
Will LGMT now but would still like to see the `overrides.less` change reverted.

**Review by delvh — APPROVED** [review]

File: .stylelintrc.yaml [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+ignoreFiles:
+  - "**/*.go"
+
 overrides:
   - files: ["**/*.less"]
     customSyntax: postcss-less
   - files: ["**/chroma/*", "**/codemirror/*", "**/standalone/*", "**/console/*"]
     rules:
       scale-unlimited/declaration-strict-value: null
+  - files: ["**/chroma/*"]
```
Comment (silverwind) [inline .stylelintrc.yaml]:
```suggestion
  - files: ["**/chroma/*", "**/codemirror/*"]
```

Fur the future in case we need it there as well.

**Comment by wxiaoguang** [comment]
> but what is this `override.less` change exactly about?

IMO it's used to provide "overwritten" styles for light/dark styles. 

"overwrite" or "base" has no different at the moment. If you like "base", I will rename it.

**Pull request merged**